nixos/modules/services/web-apps/jitsi-meet.nix

```diff
@@ -0,0 +1,497 @@
+        '';
+      };
+    };
+
+    nginx.enable = mkOption {
+      type = bool;
+      default = true;
+      description = ''
+        Whether to enable nginx virtual host that will serve the javascript application and act as
+        a proxy for the XMPP server. Further nginx configuration can be done by adapting
+        <option>services.nginx.virtualHosts.&lt;hostName&gt;</option>. It is highly recommended to
+        enable the <option>enableACME</option> and <option>forceSSL</option> options:
+
+        <programlisting>
+        services.nginx.virtualHosts.''${config.services.jitsi-meet.hostName} = {
+          enableACME = true;
+          forceSSL = true;
+        };
+        </programlisting>
+      '';
+    };
+
+    prosody.enable = mkOption {
+      type = bool;
+      default = true;
+      description = ''
+        Whether to configure Prosody to relay XMPP messages between Jitsi Meet components. Turn this
+        off if you want to configure it manually.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.jitsi-meet.jicofo.config = mapAttrs (_: v: mkDefault v) {
+      "org.jitsi.jicofo.BRIDGE_MUC" = "jvbbrewery@internal.${cfg.hostName}";
+    };
+
+    services.prosody = mkIf cfg.prosody.enable {
+      enable = mkDefault true;
+      modules = {
+        admin_adhoc = mkDefault false;
+        bosh = mkDefault true;
+        ping = mkDefault true;
+        roster = mkDefault true;
+        saslauth = mkDefault true;
+        tls = mkDefault true;
+      };
+      extraModules = [ "pubsub" ];
+      extraConfig = ''
+        certificates = "${config.services.prosody.dataDir}"
+
+        Component "conference.${cfg.hostName}" "muc"
+          storage = "memory"
+          -- modules: muc_meeting_id muc_domain_mapper
+
+          muc_room_locking = false
+          muc_room_default_public_jids = true
+
+        Component "internal.${cfg.hostName}" "muc"
+          storage = "memory"
+          -- muc_room_cache_size = 1000
+          admins = { "focus@auth.${cfg.hostName}", "jvb@auth.${cfg.hostName}" }
+
+        Component "focus.${cfg.hostName}"
+          component_secret = os.getenv("JICOFO_COMPONENT_SECRET")
+      '';
+      virtualHosts.${cfg.hostName} = {
+        enabled = true;
+        domain = cfg.hostName;
+        extraConfig = ''
+          authentication = "anonymous"
+          c2s_require_encryption = false
+          admins = { "focus@auth.${cfg.hostName}" }
+        '';
+      };
+      virtualHosts."auth.${cfg.hostName}" = {
+        enabled = true;
+        domain = "auth.${cfg.hostName}";
+        extraConfig = ''
+          authentication = "internal_plain"
+        '';
+      };
+    };
+    systemd.services.prosody.serviceConfig = mkIf cfg.prosody.enable {
+      EnvironmentFile = [ "${dataDir}/secrets" ];
+      SupplementaryGroups = [ "jitsi-meet" ];
+    };
+
+    users.groups.jitsi-meet = {};
+    systemd.tmpfiles.rules = [
+      "d '${dataDir}' 0750 root jitsi-meet - -"
+    ];
```

Comment:
We can use `StateDirectory` and `StateDirectoryMode` inside one or multiple units to set this up, as it's relative to `/var/lib`.

Reply:
I don't think so, the `jitsi-meet-init-secrets` systemd unit needs the directory and it is run before the other services from this module.